ts/importFlake.ts

```diff
@@ -0,0 +1,63 @@
+import { Check } from "./check.ts";
+import { Environment, mkEnvironment } from "./environment.ts";
+import { Executable, mkExecutable } from "./executable.ts";
+import { NixExpression, nixFlakeDep, nixRaw, nixStrLit } from "./nix.ts";
+import { mkPackage, Package } from "./package.ts";
+
+function getFlake(flakeDir: string): NixExpression {
+  if (flakeDir.match(/^[.][.]?[/]/)) {
+    const callFlake = nixFlakeDep("call-flake", {
+      url: "github:divnix/call-flake",
+    });
+    return nixRaw`(${callFlake} ${nixRaw(flakeDir)})`;
+  }
+  return nixFlakeDep(flakeDir.replaceAll(/[^a-zA-Z0-9]/g, "-"), {
+    url: flakeDir,
+  });
+}
+
+export function importFlake(flakeDir: string): {
```

Comment:
```suggestion
export function importFlake(flakeUrl: string): {
```
Since this may be a github flake url.